x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -192,9 +192,15 @@ export const InnerWorkspacePanel = React.memo(function InnerWorkspacePanel({
     dataViews,
   };
 
+  const initialRenderTime = useRef<number>(performance.now());
+  const dataReceivedTime = useRef<number>(0);
+
   const onRender$ = useCallback(() => {
-    const startTime = window.performance.now();
     if (renderDeps.current) {
+      console.log(
+        'visualization took to render after data received',
```

Comment:
Time for data to arrive

Reply:
This one won't be accurate until we have a solution to https://github.com/elastic/elastic-charts/issues/2124

Reply:
Temporary fix added: https://github.com/elastic/kibana/pull/163002/files#diff-8282b1418609c26c34a21cafe77c7da9064f898138a144ffe4aa07159fd153efR312-R315